habitat_sim/robots/mobile_manipulator.py

```diff
@@ -0,0 +1,464 @@
+            0,  # velocity_target
+            self.params.arm_mtr_vel_gain,  # velocity_gain
+            self.params.arm_mtr_max_impulse,  # max_impulse
+        )
+        self.sim_obj.create_all_motors(jms)
+        self._update_motor_settings_cache()
+
+        # set correct gains for wheels
+        if self.params.wheel_joints is not None:
+            jms = JointMotorSettings(
+                0,  # position_target
+                self.params.wheel_mtr_pos_gain,  # position_gain
+                0,  # velocity_target
+                self.params.wheel_mtr_vel_gain,  # velocity_gain
+                self.params.wheel_mtr_max_impulse,  # max_impulse
+            )
+            # pylint: disable=not-an-iterable
+            for i in self.params.wheel_joints:
+                self.sim_obj.update_joint_motor(self.joint_motors[i][0], jms)
+
+        # set initial states and targets
+        self.arm_joint_pos = self.params.arm_init_params
+        self.gripper_joint_pos = self.params.gripper_init_params
+
+        self._update_motor_settings_cache()
+
+    def update(self) -> None:
+        """Updates the camera transformations and performs necessary checks on
+        joint limits and sleep states.
+        """
+        agent_node = self._sim._default_agent.scene_node
+        inv_T = agent_node.transformation.inverted()
+
+        # Update the cameras
+        for sensor_name in self._head_sensor_names:
+            sens_obj = self._sim._sensors[sensor_name]._sensor_object
+            head_T = self._get_head_cam_transform()
+            sens_obj.node.transformation = inv_T @ head_T
+
+        for sensor_name in self._arm_sensor_names:
+            sens_obj = self._sim._sensors[sensor_name]._sensor_object
+            arm_T = self._get_arm_cam_transform()
+            sens_obj.node.transformation = inv_T @ arm_T
+
+        # Guard against out of limit joints
+        # TODO: should auto clamping be enabled instead? How often should we clamp?
+        if self._limit_robo_joints:
+            self.sim_obj.clamp_joint_limits()
+
+        self.sim_obj.awake = True
+
+    def reset(self) -> None:
+        """Reset the joints on the existing robot.
+        NOTE: only arm and gripper joint motors (not gains) are reset by default, derived class should handle any other changes."""
+
+        # reset the initial joint positions
+        self.arm_joint_pos = self.params.arm_init_params
+        self.gripper_joint_pos = self.params.gripper_init_params
+
+        self._update_motor_settings_cache()
+
+    #############################################
+    # ARM RELATED
+    #############################################
+    @property
+    def arm_joint_limits(self) -> Tuple[np.ndarray, np.ndarray]:
+        """Get the arm joint limits in radians"""
+        arm_pos_indices = list(
+            map(lambda x: self.joint_pos_indices[x], self.params.arm_joints)
+        )
+
+        lower_lims = [self.joint_limits[0][i] for i in arm_pos_indices]
```

Comment:
```suggestion
lower_lims, upper_lims = self.joint_limits[:, arm_pos_indices]
```
Will be, much, much faster.